website/content/docs/connect/observability/grafanadashboards/consulserverdashboard.mdx

```diff
@@ -0,0 +1,96 @@
+
+- **Grafana query:** `rate(consul_dns_domain_query_count[5m])`
+- **Description:** This metric tracks the rate of DNS queries per node, bucketed into 5-minute intervals. It helps monitor the query load on Consul’s DNS service.
+
+### DNS domain query time
+
+- **Grafana query:** `consul_dns_domain_query`
+- **Description:** Measures the time spent handling DNS domain queries. Spikes in this metric may indicate high contention in the catalog or too many concurrent queries.
+
+### DNS reverse query time
+
+- **Grafana query:** `consul_dns_ptr_query`
+- **Description:** Tracks the time spent processing reverse DNS queries. Spikes in query time may indicate performance bottlenecks or increased workload.
+
+### KV applies per 5 minutes
+
+- **Grafana query:** `rate(consul_kvs_apply_count[5m])`
+- **Description:** This metric tracks the rate of Key-Value store applies over 5-minute intervals, indicating the operational load on Consul’s KV store.
+
+### KV apply time
+
+- **Grafana query:** `consul_kvs_apply`
+- **Description:** Measures the time taken to apply updates to the Key-Value store. Spikes in this metric might suggest resource contention or client overload.
+
+### Transaction apply time
+
+- **Grafana query:** `consul_txn_apply`
+- **Description:** Tracks the time spent applying transaction operations in Consul, providing insights into potential bottlenecks in transactional workloads.
+
+### ACL resolves per 5 minutes
+
+- **Grafana query:** `rate(consul_acl_ResolveToken_count[5m])`
+- **Description:** This metric tracks the rate of ACL token resolutions per 5-minute intervals. It provides insights into the activity related to ACL tokens within the cluster.
+
+### ACL resolve token time
+
+- **Grafana query:** `consul_acl_ResolveToken`
+- **Description:** Measures the time taken to resolve ACL tokens into their associated policies. Spikes in this metric might indicate resource issues or configuration problems.
+
+### ACL updates per 5 minutes
+
+- **Grafana query:** `rate(consul_acl_apply_count[5m])`
+- **Description:** Tracks the rate of ACL updates per 5-minute intervals. This metric helps monitor changes in ACL configurations over time.
+
+### ACL apply time
+
+- **Grafana query:** `consul_acl_apply`
+- **Description:** Measures the time spent applying ACL changes. Spikes in apply time might suggest resource constraints or high operational load.
+
+### Catalog operations per 5 minutes
+
+- **Grafana query:** `rate(consul_catalog_register_count[5m])`, `rate(consul_catalog_deregister_count[5m])`
+- **Description:** Tracks the rate of register and deregister operations in the Consul catalog, providing insights into the churn of services within the cluster.
+
+### Catalog operation time
+
+- **Grafana query:** `consul_catalog_register`, `consul_catalog_deregister`
+- **Description:** Measures the time taken to complete catalog register or deregister operations. Spikes in this metric can indicate performance issues within the catalog.
```

Comment:
```suggestion
- **Description:** Measures the time taken to complete catalog register or deregister operations.
```
Spikes in these values just mean that a large number of services were registered, or deregistered. It does not necessarily mean that there is a performance issue.